packages/adapter-components/src/elements/ducktype/type_elements.ts

```diff
@@ -242,6 +251,7 @@ export const generateType = ({
     fields,
     path,
     isSettings: transformationConfigByType[naclName]?.isSingleton ?? false,
+    annotations: hideTypes ? ({ [CORE_ANNOTATIONS.HIDDEN]: hideTypes }) : {},
```

Comment:
nit: maybe 
```suggestion
    annotations: hideTypes ? ({ [CORE_ANNOTATIONS.HIDDEN]: true }) : undefined,
```
(explicitly use `true` just in case, and don't assume the default annotations will always remain empty, leave that up to the constructor)